File: python/mail.py
#!/usr/bin/python3
 
import smtplib
from email.mime.text import MIMEText
from email.header import Header
 
# 第三方 SMTP 服务
mail_host="smtp.sohu.com"  #设置服务器
mail_user="fxqyem@sohu.com"    #用户名
mail_pass="GLDYJZQ132R"   #口令 
 
 
sender = 'fxqyem@sohu.com'
receivers = ['cectdwf@126.com']  # 接收邮件
 
message = MIMEText('邮件发送测试...<br/><img src="http://img.netbian.com/file/2019/1221/541e78ce236d7ffb10b8d1348cbc70d6.jpg"/><br/>', 'html', 'utf-8')
message['From'] = Header("fxqy", 'utf-8')
message['To'] =  Header("dwf", 'utf-8')
 
subject = 'Python SMTP 邮件测试'
message['Subject'] = Header(subject, 'utf-8')
 
 
try:
    smtpObj = smtplib.SMTP()
    smtpObj.connect(mail_host, 25)   # 25 为 SMTP 端口号
    smtpObj.login(mail_user,mail_pass)
    smtpObj.sendmail(sender, receivers, message.as_string())
    print ("邮件发送成功")
except smtplib.SMTPException:
    print ("Error: 无法发送邮件")

input("Press any key to close...")

File: perl/mail.pl
#!/usr/bin/perl -w
use strict;
use Net::SMTP;
use Authen::SASL;

##BEGIN {
##push (@INC,'/usr/share/perl5/vendor_perl/Net/ ');
##}

my $from = 'fxqyem@sohu.com';
my $to = 'cectdwf@126.com';
my $username = 'fxqyem@sohu.com';
my $passwd = 'GLDYJZQ132R';
my $subject = 'Test Email';
my $smtp = Net::SMTP->new('smtp.sohu.com');

$smtp->auth($username,$passwd) or die "Could not authenticate $!";
$smtp->mail($from);
$smtp->to($to);

$smtp->data();
$smtp->datasend("From:$from\n");
$smtp->datasend("To: $to\n");
#$smtp->datasend("Cc: $Cc/n");
$smtp->datasend("Subject: $subject\n");
$smtp->datasend("Headers: MIME-Version: 1.0\r\nContent-type: text/html\n");
$smtp->datasend("This is a test perl email, It is successfull");
$smtp->datasend("\n<a href='http://so.com'>搜索</a>");
$smtp->datasend("\n<img src='https://p3.ssl.qhimgs1.com/sdr/400__/t01c2c8eead9c6c516d.jpg'/>");
$smtp->dataend();

$smtp->quit;

print "Press any key to close...";
<STDIN>;